stripe/src/main/java/com/stripe/android/payments/StripeBrowserLauncherViewModel.kt

```diff
@@ -3,19 +3,62 @@ package com.stripe.android.payments
 import android.app.Application
 import android.content.Intent
 import android.net.Uri
+import androidx.browser.customtabs.CustomTabColorSchemeParams
+import androidx.browser.customtabs.CustomTabsIntent
 import androidx.lifecycle.ViewModel
 import androidx.lifecycle.ViewModelProvider
 import com.stripe.android.AnalyticsEvent
 import com.stripe.android.PaymentConfiguration
+import com.stripe.android.R
 import com.stripe.android.auth.PaymentBrowserAuthContract
 import com.stripe.android.networking.AnalyticsRequestExecutor
 import com.stripe.android.networking.AnalyticsRequestFactory
 
 internal class StripeBrowserLauncherViewModel(
     private val analyticsRequestExecutor: AnalyticsRequestExecutor,
-    private val analyticsRequestFactory: AnalyticsRequestFactory
+    private val analyticsRequestFactory: AnalyticsRequestFactory,
```

Comment:
If we have these as parameters only for test, maybe we should have a protected constructor that takes any arguments that are just parameterized for test.   The production code constructor could call the protected one.   Usually test code exists in the same package that is being constructed.

Not sure if that will work here or not.  But being able to understand if something is really parameterized is helpful.   Reminds me of a bug I worked on earlier this week.

Reply:
Hm we use these in non-test code as well. Can you show me an example of what you had in mind?

Reply:
I will put one together.